### PR TITLE
fix: Minimize skipping tests

### PR DESCRIPTION
**Pull request by chrisleewilcox**
## **Description**
Running tests based on TEST_SUITE only requires skipping a lot of tests which takes significant time for the testrun to complete.  Isolating tests suite within the tests folder will only skip tests within that folder significantly reducing tests run times.

This change applies to SmokeConfirmations and SmokeSwaps.  SmokeCore tests are located in multiple folders and will still skip tests that are not SmokeCore.

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

### **Before**
image

image

### **After**
image

image

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -524,6 +528,9 @@ workflows:
           inputs:
             - content: |-
                 #!/usr/bin/env bash
+                if [ -n "$TEST_SUITE_FOLDER" ]; then
+                echo "TEST_SUITE_FOLDER value is: $TEST_SUITE_FOLDER"
```
Comment (leotm) [inline bitrise.yml:532]:
```suggestion
                  echo "TEST_SUITE_FOLDER value is: $TEST_SUITE_FOLDER"
```

File: bitrise.yml [hunk 2]
```diff
@@ -678,6 +685,9 @@ workflows:
...
```
Comment (leotm) [inline bitrise.yml:689]:
```suggestion
                  echo "TEST_SUITE_FOLDER value is: $TEST_SUITE_FOLDER"
```

File: bitrise.yml
Comment (leotm) [inline bitrise.yml:1]: as follow-up perhaps worth extracting

```yml
                if [ -n "$TEST_SUITE_FOLDER" ]; then
                  echo "TEST_SUITE_FOLDER value is: $TEST_SUITE_FOLDER"
                fi                
                if [ "$TEST_SUITE" = "Regression" ]; then
                  TEST_SUITE="Regression"
                else
                  TEST_SUITE="Smoke"
                fi
                if [ -n "$TEST_SUITE_TAG" ]; then
                  echo "TEST_SUITE_TAG value is: $TEST_SUITE_TAG"
                  TEST_SUITE=$TEST_SUITE_TAG
                fi
                export METAMASK_ENVIRONMENT='local'
                export METAMASK_BUILD_TYPE='main'
                IGNORE_BOXLOGS_DEVELOPMENT="true"
                FORCE_BUNDLING=true
```

to its own script, to reduce the repetition 4x when calling
- test:e2e:android:bitrise:build
- yarn test:e2e:android:bitrise:run
- test:e2e:ios:bitrise:build
- test:e2e:ios:bitrise:run

**Review by leotm — DISMISSED** [review]
🚀

**Review by leotm — APPROVED** [review]
should be good to update branch now ^

File: bitrise.yml
Comment (chrisleewilcox) [reply bitrise.yml:1]: Thanks @leotm will bring this up to the team and create an issue for scripting portions of this file.

**Pull request merged**